keystorage/hashicorp/src/main/java/tech/pegasys/signing/hashicorp/HashicorpConnection.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.signing.hashicorp;
+
+import static io.vertx.core.http.HttpMethod.GET;
+
+import tech.pegasys.signing.hashicorp.config.KeyDefinition;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+import javax.net.ssl.SSLException;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientResponse;
+
+public class HashicorpConnection {
```

Comment:
Name of this class doesn't quite seem right to me. The methods are doing anything with the connection, there is no connect or disconnect. Thinking it's just the HashicorpVault?